src/main/java/org/opensearch/neuralsearch/processor/ExplanationResponseProcessor.java

```diff
@@ -111,8 +111,9 @@ public SearchResponse processResponse(
                         );
                     }
                     // Create and set final explanation combining all components
+                    Float finalScore = Float.isNaN(searchHit.getScore()) ? 0.0f : searchHit.getScore();
```

Comment:
When sorting by a field other than `score`:
1. `searchHit.score` will be `Float.NaN`
2. `score` in the search hit and in actual response will be `null`

We can't pass `null` as a value for the explanation object. Therefore, I've set it to `0.0` in these cases. This ensures that we always have a valid numeric value for the explanation, even when the score isn't the primary sorting factor.